inference-engine/src/legacy_api/src/convert_function_to_cnn_network.cpp

```diff
@@ -769,6 +769,17 @@ std::shared_ptr<CNNNetworkImpl> convertFunctionToICNNNetwork(const std::shared_p
         cnnLayer->insData.resize(inputCount);
 
         for (size_t i = 0; i < layer->get_output_size(); i++) {
+            // Memory node with index = 1 has no inputs according to the specification.
+            // For proper conversion, we must cut off all the layers and data nodes above ReadValue,
+            // if they are connected only with this layer.
+            bool all_to_read_value = !layer->get_output_inputs(i).empty();
+            for (const auto& output_input : layer->get_output_inputs(i)) {
+                all_to_read_value
+                    &= std::dynamic_pointer_cast<ngraph::op::ReadValue>(output_input->get_node()) != nullptr;
+            }
+            if (all_to_read_value)
```

Comment:
Do we need to check it for every type of node?

Reply:
Most likely it will be Constant, but not guaranteed.